### PR TITLE
Fix license header in a few new files

### PR DESCRIPTION
**Pull request by am11**
Some new code files have brought back the old (three-liner) header since https://github.com/dotnet/runtime/commit/3b6600cbefe6b96706c5315ce184accbb4501f3b was merged.

cc @stephentoub

### CODE REVIEW

File: src/coreclr/tools/Common/Compiler/Logging/ReferenceSource/CompilerGeneratedState.cs [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/Common/Compiler/Logging/ReferenceSource/CompilerGeneratedState.cs:3]: Files under ReferenceSource are literal copies from the dotnet/linker repo. It doesn't make much sense updating them here. The change will be stomped over next time we do integration.

Comment (am11) [reply src/coreclr/tools/Common/Compiler/Logging/ReferenceSource/CompilerGeneratedState.cs]: Yeah, I've rolled it back.

Comment (am11) [reply src/coreclr/tools/Common/Compiler/Logging/ReferenceSource/CompilerGeneratedState.cs:3]: Updated license upstream as well https://github.com/dotnet/linker/pull/2303.

**Review by akoeplinger — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]
Thanks!

**Pull request merged**